src/journal-remote/journal-upload.c

```diff
@@ -121,6 +133,108 @@ static int check_cursor_updating(Uploader *u) {
         return 0;
 }
 
+int config_parse_compression(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Compression* compression = ASSERT_PTR(data);
+
+        assert(filename);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                *compression = COMPRESSION_NONE;
+        } else {
+                *compression = compression_from_string(ascii_strupper(strdup(rvalue)));
```

Comment:
1. duplicated string is leaked,
2. missing error check in compression_from_string(),
3. do not assign unsupported type.
```c
_cleanup_free_ char *s = strdup(rvalue);
if (!s)
        return log_oom();

Compression c = compression_from_string(ascii_strupper(s));
if (c < 0)
        return log_syntax_parse_error(...);
if (!compression_supported(c))
        return log_syntax(unit, ...);

*compression = c;
return 0;
```